aikau/src/main/resources/alfresco/forms/controls/NumberSpinner.js

```diff
@@ -161,30 +224,59 @@ define(["alfresco/forms/controls/BaseFormControl",
        */
       isNumberValidator: function alfresco_forms_controls_FormControlValidationMixin__isNumberValidator(validationConfig) {
          var isValid = false;
-         try {
-            var value = this._removeCommasAndSpaces(this.wrappedWidget.textbox.value),
-               isNumber = this._valueIsNumber(value);
-            isValid = isNumber || (this.permitEmpty && value === "");
-         } catch (e) {
+         try 
+         {
+            var textBoxValue = this.wrappedWidget.textbox.value;
+            var parsed = number.parse(textBoxValue);
+            isValid = !isNaN(parsed) || (this.permitEmpty && textBoxValue === "");
+         }
+         catch (e) 
+         {
             this.alfLog("warn", "Error validating number: ", e);
          }
          this.reportValidationResult(validationConfig, isValid);
       },
 
       /**
-       * This validator checks that the value has no more than the specified number of decimal places
+       * This validator checks that the value has no more than the specified number of decimal places (or isn't a number!)
        *
        * @instance
        * @param {object} validationConfig The configuration for this validator
        */
       decimalPlacesValidator: function alfresco_forms_controls_FormControlValidationMixin__decimalPlacesValidator(validationConfig) {
-         var isValid = false;
-         try {
-            var value = this._removeCommasAndSpaces(this.wrappedWidget.textbox.value),
-               isNumber = this._valueIsNumber(value);
-               numDecimals = value.indexOf(".") === -1 ? 0 : value.split(".")[1].length;
-            isValid = !isNumber || numDecimals === this.permittedDecimalPlaces;
-         } catch (e) {
+         var isValid = true;
+         try 
+         {
+            // Get the raw value of the number box (for locales that use the "," for a decimal place, then the value will include
+            // the ",")...
+            var textBoxValue = this.wrappedWidget.textbox.value;
+
+            // Parse the number (this will convert the "," to ".")...
+            var parsed = number.parse(textBoxValue);
+            if (isNaN(parsed))
+            {
+               // Can't do anything with a value that isn't a number - it should have been picked up by the "isNumberValidator" anyway
```

Comment:
I'm not sure it's necessary here... this validator is only testing the number of decimal places - that's it's only requirement. It's testing that the permitted number of decimal places has **not been exceeded**. So an invalid number error will have been picked up elsewhere and isn't really pertinent to this validation test (bear in mind that this is just one of a set of validations that are being performed).